bazel/format/clang_tidy/clang_tidy.bzl

```diff
@@ -158,11 +137,11 @@ def _clang_tidy_aspect_impl(target, ctx):
             src,
             target.label.name,
         )
-        for src in srcs
+        for src in _rule_sources(ctx)
```

Comment:
I think this outputs can be one run i guess? i don't see any reason why we cannot run clang-tidy for all source codes in the target at once vs here run each file per clang-tidy.

Reply:
let me test ...

Reply:
so the one issue with this plan is that the flags are set according to whether `src` is `.c` or `.cc`

a quick grep of the code suggests there are not any builds that have both as a src (not sure that would make sense - but what do i know)

on the upsides it makes the code cleaner and removes the need to use `to_list` on the srcs

Reply:
i believe all Envoy codes are C++ hence .cc and `.c` are for test codes, so we can simply ignore them

Reply:
so next problem is 

```console
 38     outfile = ctx.actions.declare_file(
 39         "bazel_clang_tidy_%s.%s.clang-tidy.yaml" % (infile.path, discriminator)
 40     )

```

i can workaround for now -  i think in the original implementation this was intended for the fix file altho not sure it was actually implemented

my intention is to use this output for the output from clang-tidy and then collect those up so probs we can think of a different path scheme

Reply:
lets deal with this separately - the benefit is marginal at best and it requires quite a lot of change to make it work